### PR TITLE
Cache

### PR DESCRIPTION
**Pull request by bkzhang**
And i guess I forgot to add topWinners in a pr.

I have part of the cache, but lots of things are broken since userHandle should now be an ID. Will add undo using cache in some other pr

### CODE REVIEW

**Review by joeyzhoucode — APPROVED** [review]
looks cool to me. Make sure you read the comments on your prev PR tho.

File: middleware/crud-functions/postWinners.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 const db = require('../../firebase.js');
 const calculateRankScore = require('../../utilities/calculations.js');
+require('../../cache.js');
```
Comment (Ipkhchan) [inline middleware/crud-functions/postWinners.js]: can we be more explicit here? const cache = require('../../cache.js');

File: cache.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+class Cache {
+  constructor() {
+    this.lastWinners = [];
+    this.lastLosers = [];
+  }
+
+  newWinners(winners) {
```
Comment (Ipkhchan) [inline cache.js]: can we name this as an action? setNewWinners

File: cache.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    this.lastWinners = winners;
+  }
+
+  newLosers(losers) {
```
Comment (Ipkhchan) [inline cache.js]: setNewLosers instead?

File: middleware/crud-functions/postWinners.js [hunk 4]
```diff
@@ -31,6 +32,7 @@ function postWinners(userHandles) {
 
   const updated = Promise.all(promises)
     .then((_result) => {
+      cache.setWinners(userHandles);
```
Comment (Ipkhchan) [inline middleware/crud-functions/postWinners.js]: this method name doesn't match the ones on the class.

Comment (Ipkhchan) [inline middleware/crud-functions/postWinners.js]: Can you also implement setting the losers in the cache in postLosers?

**Review by Ipkhchan — CHANGES_REQUESTED** [review]

File: cache.js (hunk 2 above)
Comment (bkzhang) [reply cache.js]: i dont think it should be setNewWinners, if we expand the size of the cache, itll be a queue which will be inserting (or some other ds), so I left it as newWinners to indicate the newest winners

File: middleware/crud-functions/postWinners.js (hunk 4 above)
Comment (bkzhang) [reply middleware/crud-functions/postWinners.js]: yikes oops

**Review by Ipkhchan — APPROVED** [review]

**Pull request merged**